frame/vamm/src/lib.rs

```diff
@@ -665,17 +661,18 @@ pub mod pallet {
 
 			match asset_type {
 				AssetType::Base => Ok(vamm_state.base_asset_twap),
-				AssetType::Quote => Ok(vamm_state.quote_asset_twap),
+				AssetType::Quote => Ok(Self::reciprocal_twap(&vamm_state.base_asset_twap)?),
```

Comment:
could use `.reciprocal()` here

Reply:
Ah, I see. That makes sense. I would suggest creating a `TryReciprocal` trait that implements a `try_reciprocal` method returning the appropriate arithmetic error upon failure. What do you think?